File: backend/book.rb
require_relative "product"

class Book < Product
  def ship(order)
    print_shipping_label(order.address.zipcode)
  end

  private

  def print_shipping_label(shipping_zipcode)
    shipping_zipcode + " Item isento de impostos conforme disposto na Constituição Art. 150, VI, d."
  end
end

File: backend-fabricio/src/main/kotlin/order/Book.kt
package order

import shipping.ShippingBoxService
import shipping.ShippingLabel

class Book(name: String, price: Double) : Product(name, price) {
    private val bookShippingLabelNotification = "Este item é isento de impostos conforme disposto na Constituição Art. 150, VI, d."

    override fun pay(payment: Payment) {
        val shippingLabel = ShippingLabel(payment.order.id.toString(), bookShippingLabelNotification)
        OrderPaymentEventContainer.addEvent(BookProductPaidEvent(payment.order, shippingLabel, ShippingBoxService()))

    }
}